source/ui/PGM/PGMPersistentView.cpp

```diff
@@ -0,0 +1,220 @@
+    connect(PGMBeamline::pgm()->exposedControlByName("BPM 11ID #1-Y"), SIGNAL(valueChanged(double)), this, SLOT(onBPM11ID1ValueYChanged(double)));
+
+    connect(PGMBeamline::pgm()->exposedControlByName("BPM 11ID #2-X"), SIGNAL(valueChanged(double)), this, SLOT(onBPM11ID2ValueXChanged(double)));
+    connect(PGMBeamline::pgm()->exposedControlByName("BPM 11ID #2-Y"), SIGNAL(valueChanged(double)), this, SLOT(onBPM11ID2ValueYChanged(double)));
+
+    connect(PGMBeamline::pgm()->exposedControlByName("Ring Current"), SIGNAL(valueChanged(double)), this, SLOT(onRingCurrentChange(double)));
+    connect(PGMBeamline::pgm()->exposedControlByName("Beam Lifetime"), SIGNAL(valueChanged(double)), this, SLOT(onLifetimeChange(double)));
+
+   // Main layout
+   QVBoxLayout *mainPanelLayout = new QVBoxLayout;
+   mainPanelLayout->addLayout(storageRingLayout);
+   mainPanelLayout->addWidget(bpmBox);
+   mainPanelLayout->addSpacerItem(new QSpacerItem(10, 10, QSizePolicy::Minimum, QSizePolicy::Expanding));
+
+   // Note: Lucia perfers her beamline to be referreed to by it's full acroynm. It's not nessisary to use
+   // the full acroynm in the code but whenever it's displayed to user please try to use VLS-PGM.
+   QGroupBox *persistentPanel = new QGroupBox("VLS-PGM Beamline");
+   persistentPanel->setLayout(mainPanelLayout);
+
+   QVBoxLayout *mainLayout = new QVBoxLayout;
+   mainLayout->addWidget(persistentPanel);
+
+   setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Expanding);
+   setMaximumHeight(1000);
+   setMinimumHeight(800);
+   setMaximumWidth(400);
+   setMinimumWidth(400);
+
+   setLayout(mainLayout);
+
+
+}
+
+void PGMPersistentView::onBPM10IDValueXChanged(double value){
+
+    if( value > (avg10IDx_ + bpmVariance_) || value < (avg10IDx_ - bpmVariance_) ){
+        bpm10IDvalueX_->setStyleSheet( "color : red"  );
+        bpm10IDvalueX_->setText(QString("%1 um").arg(value, 0, 'f', 0));
+    } else {
+        bpm10IDvalueX_->setStyleSheet( "color : black"  );
+        bpm10IDvalueX_->setText(QString("%1 um").arg(value, 0, 'f', 0));
+    }
+
+
+}
+
+void PGMPersistentView::onBPM10IDValueYChanged(double value){
+
+    if( value > (avg10IDy_ + bpmVariance_) || value < (avg10IDy_ - bpmVariance_) ){
+        bpm10IDvalueY_->setStyleSheet( "color : red"  );
+        bpm10IDvalueY_->setText(QString("%1 um").arg(value, 0, 'f', 0));
+    } else {
+        bpm10IDvalueY_->setStyleSheet( "color : black"  );
+        bpm10IDvalueY_->setText(QString("%1 um").arg(value, 0, 'f', 0));
+    }
+}
+
+void PGMPersistentView::onBPM11ID1ValueXChanged(double value){
+
+    if( value > (avg11ID1x_ + bpmVariance_) || value < (avg11ID1x_ - bpmVariance_) ){
+        bpm11ID1valueX_->setStyleSheet( "color : red"  );
+        bpm11ID1valueX_->setText(QString("%1 um").arg(value, 0, 'f', 0));
+    } else {
+        bpm11ID1valueX_->setStyleSheet( "color : black"  );
+        bpm11ID1valueX_->setText(QString("%1 um").arg(value, 0, 'f', 0));
```

Comment:
In fact, all these bpm controls share a commonality.
* They have an average value and a variance.
* Getting variance distance from the average puts them in a bad state
* Otherwise they are in a good state.

I think it might be an idea to wrap these controls in a class, which contains the logic for determining whether they are in a good state or not. Then all the view needs to do is listen for a 'inValidStateChanged(bool)' type signal and update itself accordingly.